src/gui/src/PatternEditor/PatternEditorPanel.cpp

```diff
@@ -884,6 +870,85 @@ void PatternEditorPanel::patternSizeChanged( int nSelected )
 	EventQueue::get_instance()->push_event( EVENT_SELECTED_PATTERN_CHANGED, -1 );
 }
 
+void PatternEditorPanel::updatePatternSizeLCD(){
+	// update pattern size
+	int nPatternSize = m_pPattern->get_length();
+	int den = m_pPattern->get_denominator();
+	int nBeatUnit = MAX_NOTES / den;
+
+	char tmp[20];
+	if (nPatternSize % nBeatUnit == 0) {
+		sprintf( tmp, "%d/%d", nPatternSize / nBeatUnit, den);
+	} 
+	else {
+		sprintf( tmp, "%.3f/%d", (float) nPatternSize / nBeatUnit, den);
+	}
+	__pattern_size_LCD->setText( tmp );//
+}
+
+void PatternEditorPanel::patternSizeLCDClicked()
+{
+	Hydrogen *pEngine = Hydrogen::get_instance();
+	if ( pEngine->getState() != STATE_READY ) {	
+		QMessageBox::information( this, "Hydrogen", tr( "Is not possible to change the pattern size when playing." ) );
+		return;
+	} //TODO is it really impossible to change the pattern size when playing?
+
+	bool bIsOkPressed;
+	int denominator;
+	
+	QString qtmp = QInputDialog::getText( this, "Hydrogen",
+				tr( "New Pattern Size\nIndicate numerator / denominator (eg. \"4/4\") or just numerator"),
+				QLineEdit::Normal, __pattern_size_LCD->getText(), &bIsOkPressed );
+	
+	if ( bIsOkPressed  ) {		
+		QStringList parts = qtmp.split('/');
+		int nDenominator = m_pPattern->get_denominator();
+		if ( parts.size() == 1 || parts.size() == 2) { // must reject if parts.size > 2 or null
+		    bool bOk;
+		    double fNumerator = parts[0].toDouble( &bOk );
+		    if ( bOk && parts.size() == 2 ) {
+			nDenominator = parts[1].toInt( &bOk );
+			if (bOk && (nDenominator <= 0 || nDenominator > 192) ){
```

Comment:
Yes

Reply:
You can use `tr()` and `QString::args()` together, but you have to use them in the right order. `tr()` needs an obvious constant arg because the translation tools parse the source. So you would want

```c++
QString( tr( "ticks per bar is %1") ).arg(MAX_NOTES)
```

but not

```c++
tr( QString("ticks per bar is %1").arg(MAX_NOTES) )
```
if you see what I mean.

Reply:
Probably yes, what about `tr( "ticks per bar is %1").arg(MAX_NOTES)` ?

Reply:
Oh yes, you're right. The return type of `QObject::tr()` is a `QString` already so no need for that `QString()`